frontend/ui/translator.lua

```diff
@@ -349,6 +351,122 @@ Returns decoded JSON table from translate server.
 @treturn string result, or nil
 --]]
 function Translator:loadPage(text, target_lang, source_lang)
+    local trans_server = self:getTransServer()
+    if trans_server == "https://fanyi.youdao.com" then
+        return self:loadPageByYoudao(text, target_lang, source_lang)
+    end
+    return self:loadPageByGoogle(text, target_lang, source_lang)
+end
+
+local function http_request(s_url, method, headers, request)
+    local socket = require('socket')
+    local socketutil = require("socketutil")
+    local http = require('socket.http')
+    local ltn12 = require("ltn12")
+
+    local sink = {}
+    if request == nil then
+        request = {}
+    end
+    socketutil:set_timeout()
+
+    request['url'] = s_url
+    request['method'] = method
+    request['sink'] = ltn12.sink.table(sink)
+    request['headers'] = headers
```

Comment:
Please use double quotes here as per the style guide. Besides that it looks good!

Reply:
Thanks, it's douban quotes now